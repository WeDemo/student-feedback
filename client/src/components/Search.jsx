import React from 'react';

import symbols from '../symbols.jsx';

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.query = React.createRef();
    this.handleInputSubmit = this.handleInputSubmit.bind(this);
  }

  handleInputSubmit(event) {
    event.preventDefault();
    this.props.searchFilter(this.query.current.value);
  }

  render() {
    return (
      <form className='reviewSearchBar' onSubmit={this.handleInputSubmit}>
        <div className='reviewSearchInputWrapper'>
            <input type='text' className='reviewSearchInput' placeholder='Search reviews' ref={this.query}/>
            <button className='reviewSearchButton' type='submit'>{symbols.magGlass}</button>
        </div>
      </form>
    );
  }
}

export default Search;
