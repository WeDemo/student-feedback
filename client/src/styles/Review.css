.individualReviewContainer {
  border-bottom: 1px solid #cdd2dd;
  padding: 30px 0px 30px 10px;
  min-height: 120px;
}

.reviewUserInfo {
  width: 35%;
  height: 100%;
  display: inline-block;
  vertical-align: top;
}

.reviewInfo {
  display: inline-block;
  padding: 3px 0 0 15px;
  vertical-align: top;
}

.reviewDate {
  color: #686f7a;
}

.reviewRatingAndText {
  width: 65%;
  height: 100%;
  display: inline-block;
  vertical-align: top;
}

.reviewStars{
  width: 110px;
  height: 17px;
  padding-bottom: 10px;
}

.reviewUserPic {
  display: inline-block;
  width: 46px;
  height: 46px;
  border-radius: 50%;
  vertical-align: top;
}

.reviewUserInitials {
  display: inline-block;
  width: 46px;
  height: 46px;
  border: 1px solid #686f7a;
  border-radius: 50%;
  text-align: center;
  color:#686f7a;
  line-height: 46px;
  vertical-align: top;
}