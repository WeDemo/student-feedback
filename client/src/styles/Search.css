
.reviewSearchBar {
  float: right;
  font-size: 18px;
  font-weight: 600;
  height: 33px;
 }
 
 .clearSearchButton {
   position: absolute;
   z-index: 1;
   right: 33px;
   height: 32px;
   background-color: transparent;
   border-style: none;
   outline: none;
   cursor: pointer;
 }
 
 .clearSearchSymbol {
   color: #8c8f92c9;
   z-index: 2;
   opacity: 1;
 }
 
 .reviewSearchButton {
   font-size: 12px;
   color: white;
   border-style: none;
   outline: none;
   background-color: #007791;
   text-align: center;
   height: 32px;
   width: 32px;
   border-radius: 2px;
   position: relative;
   left: -2px;
   cursor: pointer;
 }
 
 .reviewSearchButton:hover {
   background-color: #014f61;
 }

 .reviewSearchInput {
  height: 28px;
  box-shadow: none;
  font-size: 12px;
  font-weight: 600;
  text-indent: 8px;
  border-radius: 2px;
  border: 1px solid #b3b9c5;
  vertical-align: top;
  width: 200px;
}
   
.reviewSearchInput:focus {
  box-shadow: 1px solid #1799b6;
  outline: 1px auto #1799b6;
  border: 1px solid #1799b6;
}