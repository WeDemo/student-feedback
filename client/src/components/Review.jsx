import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import styles from '../styles/Review.css';

class Review extends React.Component {
  setReviewText() {
    const { review } = this.props.reviewData;
    if (typeof review === 'object') {
      return (
        <div>
          {review.preQuery}
          {review.query}
          {review.postQuery}
        </div>
      );
    }
    return (
      <div>
        {review}
      </div>
    );
  }

  setUserPic() {
    const { userPic } = this.props.reviewData.user;
    if (userPic.includes('https')) {
      return <img className={styles.reviewUserPic} src={userPic}></img>;
    }
    return <div className={styles.reviewUserInitials}>{userPic}</div>;
  }

  render() {
    return (
      <div className={styles.individualReviewContainer}>
        <div className={styles.reviewUserInfo}>
          {this.setUserPic()}
          <span className={styles.reviewInfo}>
            <div className={styles.reviewDate}>{moment(this.props.reviewData.date).fromNow()}</div>
            <div>{this.props.reviewData.user.username}</div>
          </span>
        </div>
        <div className={styles.reviewRatingAndText}>
          <div><img className={styles.reviewStars} src={`https://s3.us-east-2.amazonaws.com/udemy-demo-tarik/${this.props.reviewData.rating}+stars+white.png`}></img></div>
          {this.setReviewText()}
        </div>
      </div>
    );
  }
}

Review.propTypes = {
  reviewData: PropTypes.shape({
    user: PropTypes.shape({
      userPic: PropTypes.string.isRequired,
      username: PropTypes.string.isRequired,
    }),
    date: PropTypes.string.isRequired,
    review: PropTypes.oneOfType([PropTypes.string.isRequired, PropTypes.object.isRequired]),
    rating: PropTypes.number.isRequired,
  }),
};

export default Review;
