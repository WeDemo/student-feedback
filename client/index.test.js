import React from 'react';
import Enzyme, {shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Search from './src/components/Search.jsx';
import Review from './src/components/Review.jsx';


Enzyme.configure({ adapter: new Adapter() })

it('search component renders', () => {
  const sampleProps = {
    searchFilter: () => {},
    currentFilter: 3,
  };

  const SearchTest = renderer.create(<Search props={sampleProps} />).toJSON();
  expect(SearchTest).toMatchSnapshot();
});

describe('Review component tests', () => {
  const reviewData = {
    user: {
      userPic: 'https://picsum.photos/200/300/?random',
      userName: 'Tarik',
    },
    date: new Date(),
    review: 'something',
    rating: '5',
  };

  it('renders', () => {

    const ReviewSnap = renderer.create(<Review props={reviewData} />).toJSON();
    expect(ReviewSnap).toMatchSnapshot();
  });
});

// const Review = props => (
//   <div className='individualReviewContainer'>
//     <div className='reviewUserInfo'>
//       <span>{set.userPic(props.reviewData.user.userPic)}</span>
//       <span className='reviewInfo'>
//         <div className='reviewDate'>{moment(props.reviewData.date).fromNow()}</div>
//         <div>{props.reviewData.user.username}</div>
//       </span>
//     </div>
//     <div className='reviewRatingAndText'>
//       <div><img className='reviewStars' src={`https://s3.us-east-2.amazonaws.com/udemy-demo-tarik/${props.reviewData.rating}+stars+white.png`}></img></div>
//       <div>{set.reviewText(props.reviewData.review)}</div>
//     </div>
//   </div>
// );

// export default Review;