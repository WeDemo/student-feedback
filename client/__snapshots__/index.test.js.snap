// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
<form
  className="reviewSearchBar"
  onSubmit={[Function]}
>
  <div
    className="reviewSearchInputWrapper"
  >
    <input
      className="reviewSearchInput"
      placeholder="Search reviews"
      type="text"
    />
    <button
      className="reviewSearchButton"
      type="submit"
    >
      <i
        className="fas fa-search"
      />
    </button>
  </div>
</form>
`;

exports[`search component renders 1`] = `
<form
  className="reviewSearchBar"
  onSubmit={[Function]}
>
  <div
    className="reviewSearchInputWrapper"
  >
    <input
      className="reviewSearchInput"
      placeholder="Search reviews"
      type="text"
    />
    <button
      className="reviewSearchButton"
      type="submit"
    >
      <i
        className="fas fa-search"
      />
    </button>
  </div>
</form>
`;
